lib/msf/core/exploit/remote/http/http_cookie_jar.rb

```diff
@@ -80,7 +84,11 @@ def parse(set_cookie_header)
                 next
               end
 
-              cookies << HttpCookie.new(name, value, attrs)
+              if attrs.is_a?(Hash)
+                cookies << HttpCookie.new(name, value, **attrs)
+              else
+                cookies << HttpCookie.new(name, value)
+              end
```

Comment:
this seems a little off to me, someone passing in attrs that isn't a hash and it just gets ignored, can it be anything other than a hash? if not maybe a guard clause or at least a warning

Reply:
See below